Source/WebCore/html/canvas/WebGLRenderingContextBase.cpp

```diff
@@ -841,25 +841,18 @@ RefPtr<ImageBuffer> WebGLRenderingContextBase::transferToImageBuffer()
     return buffer;
 }
 
-void WebGLRenderingContextBase::reshape(int width, int height, int oldWidth, int oldHeight)
+void WebGLRenderingContextBase::reshape()
 {
     if (isContextLost())
         return;
 
-    if (width == oldWidth && height == oldHeight)
+    auto newSize = clampedCanvasSize();
```

Comment:
`clampedCanvasSize()` uses `canvasBase().width()` and `canvasBase().height()`. Maybe `CanvasRenderingContext` needs to have `size()`, `width()`, `height()` and `clampedCanvasSize()` and all the contexts use these methods instead of having every one implementing them individually.

Reply:
> clampedCanvasSize() uses canvasBase().width() and canvasBase().height(). 

It should only use canvasBase().size().

> Maybe CanvasRenderingContext needs to have size(), width(), height() and clampedCanvasSize() and all the contexts use these methods instead of having every one implementing them individually.

clampedCanvasSize() is WebGL specific.

size(), width(), height() are not rendering context properties per se. Each particular rendering context has internal sizes, related to the logic of their behavior. Like WebGL has default framebuffer size. WebGPU has the target texture size, etc.